gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/KafkaAvroJobStatusMonitor.java

```diff
@@ -165,6 +179,30 @@ public GobblinTrackingEvent deserializeEvent(DecodeableKafkaRecord<byte[],byte[]
         break;
       case TimingEvent.JOB_COMPLETION_PERCENTAGE:
         properties.put(TimingEvent.JOB_LAST_PROGRESS_EVENT_TIME, properties.getProperty(TimingEvent.METADATA_END_TIME));
+        break;
+      case JobEvent.WORK_UNITS_CREATED:
+        Long numWorkUnits = Long.parseLong(properties.getProperty(CountEventBuilder.COUNT_KEY));
+        String workUnitCountName = MetricRegistry.name(ServiceMetricNames.GOBBLIN_SERVICE_PREFIX,
+            properties.getProperty(TimingEvent.FlowEventConstants.FLOW_GROUP_FIELD),
+            properties.getProperty(TimingEvent.FlowEventConstants.FLOW_NAME_FIELD),
+            JobEvent.WORK_UNITS_CREATED);
+
+        MetricFilter filter = new MetricFilter() {
+          @Override public boolean matches(String name, Metric metric) {
+            return name.equals(workUnitCountName);
+          }
+        };
+        SortedMap<String, Gauge> existingGauges = this.getMetricContext().getGauges(filter);
```

Comment:
*nit
SortedMap<String, Gauge> existingGauges = this.getMetricContext().getGauges((name, metric) -> name.equals(workUnitCountName));